kratos/python_scripts/auxiliar_methods_adaptative_remeshing.py

```diff
@@ -0,0 +1,112 @@
+        self.analysis = analysis
+
+    def AdaptativeRemeshingDetectBoundary(self):
+        """This function detects the boundary to preserve pure node BC
+
+            Keyword arguments:
+            self It signifies an instance of a class.
+        """
+        solver = self.analysis._GetSolver()
+        computing_model_part = solver.GetComputingModelPart()
+        if not self.analysis.process_remesh:
+            convergence_criteria = solver.get_convergence_criterion()
+            convergence_criteria.Initialize(computing_model_part)
+
+        # Ensuring to have conditions on the BC before remesh
+        is_surface = False
+        for elem in computing_model_part.Elements:
+            geom = elem.GetGeometry()
+            if geom.WorkingSpaceDimension() != geom.LocalSpaceDimension():
+                is_surface = True
+            break
+
+        if not is_surface:
+            list_model_parts = []
+            # We need to detect the conditions in the boundary conditions
+            if self.analysis.project_parameters.Has("constraints_process_list"):
+                constraints_process_list = self.analysis.project_parameters["constraints_process_list"]
+                for i in range(0,constraints_process_list.size()):
+                    item = constraints_process_list[i]
+                    list_model_parts.append(item["Parameters"]["model_part_name"].GetString())
+            skin_detection_parameters = KratosMultiphysics.Parameters("""
+            {
+                "list_model_parts_to_assign_conditions" : []
+            }
+            """)
+            root_model_part_name = computing_model_part.GetRootModelPart().Name
+            for name in list_model_parts:
+                name = name.replace(root_model_part_name + ".", "")
+                skin_detection_parameters["list_model_parts_to_assign_conditions"].Append(name)
+
+            if computing_model_part.ProcessInfo[KratosMultiphysics.DOMAIN_SIZE] == 2:
+                detect_skin = KratosMultiphysics.SkinDetectionProcess2D(computing_model_part, skin_detection_parameters)
+            else:
+                detect_skin = KratosMultiphysics.SkinDetectionProcess3D(computing_model_part, skin_detection_parameters)
+            detect_skin.Execute()
+        solver.SetEchoLevel(self.analysis.echo_level)
+
+    def SPRAdaptativeRemeshingRunSolutionLoop(self):
+        """This function executes the solution loop of the AnalysisStage for cases where remeshing may be considered with SPR convergence criteria
+
+            Keyword arguments:
+            self It signifies an instance of a class.
+        """
+
+        # Remeshing adaptively
+        solver = self.analysis._GetSolver()
+        computing_model_part = solver.GetComputingModelPart()
+        convergence_criteria = solver.get_convergence_criterion()
+        builder_and_solver = solver.get_builder_and_solver()
+        mechanical_solution_strategy = solver.get_mechanical_solution_strategy()
```

Comment:
This is an implementational detail!

Reply:
¿?, I don't understand the issue

Reply:
this function is not in the Interface of `PythonSolver`

Reply:
You mean we removed GetComputingModelPart?

Reply:
GetComputingModelPart exists in the base solver, line 154

Reply:
but e.g. `get_mechanical_solution_strategy` is not

Reply:
OK, I will move this to structural auxiliar utilities